src/lib/stitching/exchange/transformers/replaceCommerceDateTimeType.ts

```diff
@@ -88,7 +95,9 @@ export class ReplaceCommerceDateTimeType implements Transform {
         }
       } else if (
         ["CommerceDateTime!", "CommerceDate!"].includes(
-          getNamedType(fieldDefinition).type.toString()
+          // FIXME: Something is really wrong here, you can't pass a field def
+          //        to getNamedType.
+          (getNamedType(fieldDefinition as any)! as any).type.toString()
```

Comment:
cc @ashkan18

Reply:
_ashkan pretending knowing what this `(getNamedType(fieldDefinition as any)! as any)` means_ : Nice!!

Reply:
Hahaha. What I meant is that the runtime part of that code violates the API and iirc got blame said you worked on that code. Perhaps it just happens to do something through the magic of 🦆-typing?

Reply:
haha yes, im pretty sure i wrote that code and im even more sure i was confused back then too!

Reply:
lol, well hopefully the improved types will help you then!